tests/NHSD.BuyingCatalogue.API.IntegrationTests/Features/Hosting/PrivateCloud/Supplier Edit Private Cloud Validation.3641.feature

```diff
@@ -0,0 +1,61 @@
+        | Summary                     | Link  | HostingModel | RequiresHSCN |
+        | A string with length of 501 | a@b.c | AZDO        | A new string |
+    Then a response status of 400 is returned
+    And the summary field value is the validation failure maxLength
+    And SolutionDetail exist
+        | Solution | SummaryDescription             | FullDescription   | Hosting                                                                                                                    |
+        | Sln1     | An full online medicine system | Online medicine 1 | { "PrivateCloud": {"Summary": "A Summary", "Link": "A Link", "HostingModel": "A hosting type", "RequiresHSCN": "A string" } |
+
+@3641
+Scenario: 2. Link exceeds the maxLength
+    When a PUT request is made to update the hosting-type-private-cloud section for solution Sln1
+        | Summary   | Link                         | HostingModel | RequiresHSCN |
+        | A summary | A string with length of 1001 | AZDO        | A new string |
+    Then a response status of 400 is returned
+    And the link field value is the validation failure maxLength
+    And SolutionDetail exist
+        | Solution | SummaryDescription             | FullDescription   | Hosting                                                                                                                    |
+        | Sln1     | An full online medicine system | Online medicine 1 | { "PrivateCloud": {"Summary": "A Summary", "Link": "A Link", "HostingModel": "A hosting type", "RequiresHSCN": "A string" } |
+
+@3641
+Scenario: 3. Hosting Type exceeds the maxLength
+    When a PUT request is made to update the hosting-type-private-cloud section for solution Sln1
+        | Summary   | Link  | HostingModel                 | RequiresHSCN |
+        | A summary | a@b.c | A string with length of 1001 | A new string |
+    Then a response status of 400 is returned
+    And the hosting-model field value is the validation failure maxLength
+    And SolutionDetail exist
+        | Solution | SummaryDescription             | FullDescription   | Hosting                                                                                                                    |
+        | Sln1     | An full online medicine system | Online medicine 1 | { "PrivateCloud": {"Summary": "A Summary", "Link": "A Link", "HostingModel": "A hosting type", "RequiresHSCN": "A string" } |
+
+@3641
+Scenario: 4. Private Cloud fields are set to null
+    When a PUT request is made to update the hosting-type-private-cloud section for solution Sln1
+        | Summary | Link | HostingModel | RequiresHSCN |
+        | NULL    | NULL | NULL        | NULL         |
+    Then a successful response is returned
+    And SolutionDetail exist
+        | Solution | SummaryDescription             | FullDescription   | Hosting |
+        | Sln1     | An full online medicine system | Online medicine 1 |         |
```

Comment:
Is it worth adding in another test for showing all of the maxlengths being set (and that the maxLength field is shown